src/concept2skierg.cpp

```diff
@@ -380,13 +380,14 @@ void concept2skierg::stateChanged(QLowEnergyService::ServiceState state) {
 #endif
 #endif
             if (virtual_device_enabled) {
-            emit debug(QStringLiteral("creating virtual bike interface..."));
+                emit debug(QStringLiteral("creating virtual bike interface..."));
```

Comment:
In this view of this class I don't see a check that firstStateChange != 0 or if there is already a virtual device.